name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/editor/TimePeriodMigration.java

```diff
@@ -0,0 +1,36 @@
+package name.abuchen.portfolio.ui.editor;
+
+import org.eclipse.e4.ui.model.application.ui.basic.MPart;
+import org.eclipse.jface.preference.PreferenceStore;
+
+import name.abuchen.portfolio.model.Client;
+
+class TimePeriodMigration
```

Comment:
We can do the migration here. Maybe we can also migrate lazily in the ClientInput.

The logic in ClientInput could be:
* if the new configuration set exists, use it
* otherwise use the reporting time periods from perference
* otherwise create the default reporting periods

Then we do not need to a create a new client version. It is just in the getReportingPeriod method in ClientInput


(This was not possible with the client filter, because with the client filter we introduce new UUIDs as a replacement for the auto-generated identifier. Plus: the client filters were used as data series in the chart. But with the reporting period we already have identifier (the code) and we do not have to migrate any configuration in the XML file itself).

Reply:
I understand what you mean. This "Lazy"-Migration is better. Thx for the idea.